app/server/github_verifier.go

```diff
@@ -89,5 +107,152 @@ func (g *githubVerifier) branchContains(branch, owner, repo, hash string) (bool,
+			g.markContains(owner, repo, *t.Commit.SHA)
+		}
+		// store the highest major version for github/codeql-action
+		// this helps check release branches later, due to their release process.
+		if isCodeQL {
+			version := t.GetName()
+			if semver.IsValid(version) && semver.Compare(version, g.codeqlActionMajor) == 1 {
+				g.codeqlActionMajor = semver.Major(version)
+			}
+		}
+	}
+	return nil
+}
+
+func (g *githubVerifier) markContains(owner, repo, sha string) {
+	commit := commit{
+		owner: owner,
+		repo:  repo,
+		hash:  strings.ToLower(sha),
+	}
+	g.cachedCommits[commit] = true
+}
+
+// check the most recent release branches, ignoring the default branch which was already checked.
+func (g *githubVerifier) checkReleaseBranches(owner, repo, hash, defaultBranch string, limit int) (bool, error) {
+	var (
+		analyzedBranches int
+		branches         []string
+		err              error
+	)
+
+	switch {
+	// special case: github/codeql-action releases all come from "main", even though the tags are on different branches
+	case owner == "github" && repo == "codeql-action":
+		branches, err = g.getCodeQLBranches()
+		if err != nil {
+			return false, err
+		}
+	default:
+		branches, err = g.getBranches(owner, repo)
+		if err != nil {
+			return false, err
+		}
+		// we may have discovered more commit SHAs from the release process
+		c := commit{
+			owner: owner,
+			repo:  repo,
+			hash:  hash,
+		}
+		if contains, ok := g.cachedCommits[c]; ok {
+			return contains, nil
+		}
+	}
+
+	for _, branch := range branches {
+		if analyzedBranches >= limit {
+			break
+		}
+		if branch == defaultBranch {
+			continue
+		}
+		analyzedBranches++
+		contains, err := g.branchContains(branch, owner, repo, hash)
+		if err != nil {
+			return false, err
+		}
+		if contains {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// returns the integer version from the expected format: "v1", "v2", "v3" ..
+func parseCodeQLVersion(version string) (int, error) {
+	if !strings.HasPrefix(version, "v") {
+		return 0, fmt.Errorf("%w: %s", errInvalidCodeQLVersion, version)
+	}
+	major, err := strconv.Atoi(version[1:])
+	if major < 1 || err != nil {
+		return 0, fmt.Errorf("%w: %s", errInvalidCodeQLVersion, version)
+	}
+	return major, nil
+}
+
+// these branches follow the releases/v3 pattern, so we can make assumptions about what they're called.
+// this should be called after g.getTags(), because it requires g.codeqlActionMajor to be set.
+func (g *githubVerifier) getCodeQLBranches() ([]string, error) {
+	if g.codeqlActionMajor == "" {
+		return nil, nil
+	}
+	version, err := parseCodeQLVersion(g.codeqlActionMajor)
+	if err != nil {
+		return nil, err
+	}
+	branches := make([]string, version)
+	// descending order (e..g releases/v5, releases/v4, ... releases/v1)
+	for i := 0; i < version; i++ {
+		branches[i] = "releases/v" + strconv.Itoa(version-i)
+	}
+	return branches, nil
+}
+
+func (g *githubVerifier) getBranches(owner, repo string) ([]string, error) {
```

Comment:
Above, the `githubVerifier` has a method - `getTags` - which gets the tags of the repository. The name `getBranches` sounds to me like it does the same as `getTags`, but it gets the commits of the releases, right? I would suggest a different name for this method to make this distinction clear such as `get(Branches/Commits)OfReleases`.

Reply:
Here's an example where it helped though:
https://api.github.com/repos/actions/upload-artifact/releases/173086981
```
 "tag_name": "v3.2.1-node20",
  "target_commitish": "v3/node20",
  "name": "v3.2.1-node20",
 ```